### ISSUE TITLE
[Bug]: STREAM_RECEIVER segfaults

### ISSUE THREAD

**Issue by vobruba-martin** [netdata/netdata#13694]
### Bug description

STREAM_RECEIVER thread segfaults from time to time. I can see days without any segfault and I can see days when it segfaults every a few minutes. Netdata crashes and is restarted after 2 minutes.

Is there any way how to debug this? All I see is this:

syslog:
```
kernel: [31678137.720256] STREAM_RECEIVER[8803]: segfault at 0 ip 00005601b143f837 sp 00007f0a0dc8e460 error 4 in netdata[5601b11cf000+769000]
```

netdata error.log
```
Terminated
Terminated
Terminated
Terminated
```

### Expected behavior

No segfaults

### Steps to reproduce

Unknown

### Installation method

kickstart.sh

### System info

```shell
Linux 4.15.0-48-generic #51-Ubuntu SMP Wed Apr 3 08:28:49 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux
/etc/lsb-release:DISTRIB_ID=Ubuntu
/etc/lsb-release:DISTRIB_RELEASE=18.04
/etc/lsb-release:DISTRIB_CODENAME=bionic
/etc/lsb-release:DISTRIB_DESCRIPTION="Ubuntu 18.04.6 LTS"
/etc/os-release:NAME="Ubuntu"
/etc/os-release:VERSION="18.04.6 LTS (Bionic Beaver)"
/etc/os-release:ID=ubuntu
/etc/os-release:ID_LIKE=debian
/etc/os-release:PRETTY_NAME="Ubuntu 18.04.6 LTS"
/etc/os-release:VERSION_ID="18.04"
/etc/os-release:VERSION_CODENAME=bionic
/etc/os-release:UBUNTU_CODENAME=bionic
```

### Netdata build info

```shell
Version: netdata v1.36.1
Configure options:  '--prefix=/usr' '--sysconfdir=/etc' '--localstatedir=/var' '--libexecdir=/usr/libexec' '--libdir=/usr/lib' '--with-zlib' '--with-math' '--with-user=netdata' '--with-bundled-protobuf' 'CFLAGS=-O2 -pipe' 'LDFLAGS='
Install type: custom
Features:
    dbengine:                   YES
    Native HTTPS:               YES
    Netdata Cloud:              YES 
    ACLK Next Generation:       YES
    ACLK-NG New Cloud Protocol: YES
    ACLK Legacy:                NO
    TLS Host Verification:      YES
    Machine Learning:           YES
    Stream Compression:         NO
Libraries:
    protobuf:                YES (bundled)
    jemalloc:                NO
    JSON-C:                  YES
    libcap:                  NO
    libcrypto:               YES
    libm:                    YES
    tcalloc:                 NO
    zlib:                    YES
Plugins:
    apps:                    YES
    cgroup Network Tracking: YES
    CUPS:                    NO
    EBPF:                    YES
    IPMI:                    NO
    NFACCT:                  NO
    perf:                    YES
    slabinfo:                YES
    Xen:                     NO
    Xen VBD Error Tracking:  NO
Exporters:
    AWS Kinesis:             NO
    GCP PubSub:              NO
    MongoDB:                 NO
    Prometheus Remote Write: NO
```

**Comment by ktsaou** [comment]
Ooops! We are very sorry for that! We are doing major changes to the internals of the agent and it seems that there is a case we didn't catch.

@MrZammler please help here. We need a backtrace to understand where the bug is. More likely `coredumpctl` will have more info.

**Comment by ilyam8** [comment]
@vobruba-martin if possible compile the Agent with gdb

> sudo CFLAGS="-Og -ggdb" ./netdata-installer.sh ...

**Comment by MrZammler** [comment]
@vobruba-martin Do let us know if you need any help with what Kostas and Ilya mentioned.

[coredumpctl](https://www.freedesktop.org/software/systemd/man/coredumpctl.html) can provide some info after the crash. If there's not enough then an install of a compiled version of Netdata with debug symbols can help.

You can also send me your `/var/log/netdata/error.log` to manolis@netdata.cloud it might contain some more information.

Thanks!

**Comment by vobruba-martin** [comment]
@MrZammler I've installed coredumpctl and I'm waiting for another crash. I will let you know then.

**Comment by CreeperFace00** [comment]
I am also encountering these crashes, it's persisted across 2 fresh Ubuntu 22.04.1 installs. I thought I had compiled with debug symbols, but it appears not. Here's the output of coredumpctl debug on the latest crash this morning:
[https://gist.github.com/CreeperFace00/68ab129c9b54c368b6cb1882e4bf5107](https://gist.github.com/CreeperFace00/68ab129c9b54c368b6cb1882e4bf5107)

**Comment by MrZammler** [comment]
Thanks @CreeperFace00 !

Based on that gist, is it possible to test for a while with ML disabled? In `netdata.conf` under the `[ml]` section, use `enabled = no` ?

**Comment by CreeperFace00** [comment]
I'll try that and let you know.

**Comment by MrZammler** [comment]
Also, if you have any core files, can you examine them with gdb? I see you have a custom install, maybe you could add in `CFLAGS` as @ilyam8 suggested, and with ulimit to unlimited should get a core dump.

Thanks for your help!

**Comment by CreeperFace00** [comment]
Hmm, well having a custom install wasn't intentional, might have been due to me compiling from source with the debug flags on, the ones I found [here](https://learn.netdata.cloud/docs/agent/daemon#debugging)

I compiled with debugging enabled again and added `* soft core unlimited` to /etc/security/limits.conf for the coredumps.

```shell
max@netdata:~$ netdata -W buildinfo
Version: netdata v1.36.0-138-g4453613cb
Configure options:  '--prefix=/usr' '--sysconfdir=/etc' '--localstatedir=/var' '--libexecdir=/usr/libexec' '--libdir=/usr/lib' '--with-zlib' '--with-math' '--with-user=netdata' '--with-bundled-protobuf' 'CFLAGS=-O1 -ggdb -DNETDATA_INTERNAL_CHECKS=1' 'LDFLAGS='
Install type: custom
Features:
    dbengine:                   YES
    Native HTTPS:               YES
    Netdata Cloud:              NO
    ACLK:                       NO
    TLS Host Verification:      YES
    Machine Learning:           NO
    Stream Compression:         YES
Libraries:
    protobuf:                YES (bundled)
    jemalloc:                NO
    JSON-C:                  YES
    libcap:                  NO
    libcrypto:               YES
    libm:                    YES
    tcalloc:                 NO
    zlib:                    YES
Plugins:
    apps:                    YES
    cgroup Network Tracking: YES
    CUPS:                    NO
    EBPF:                    YES
    IPMI:                    NO
    NFACCT:                  NO
    perf:                    YES
    slabinfo:                YES
    Xen:                     NO
    Xen VBD Error Tracking:  NO
Exporters:
    AWS Kinesis:             NO
    GCP PubSub:              NO
    MongoDB:                 NO
    Prometheus Remote Write: NO
```

I'll keep an eye on it and see if it happens again. Yesterday it went nearly 24 hours without a crash, which is the longest I've seen my setup run without issues in a while.

**Comment by MrZammler** [comment]
@CreeperFace00 after #13707 is it possible to check with tommorow's nightly that includes that patch, and ML enabled?

**Comment by CreeperFace00** [comment]
I did a packet capture, and interestingly the child nodes are still communicating and sending metrics to the parent node as normal, even with the dashboard inaccessible.

**Comment by CreeperFace00** [comment]
Ah, it did generate a coredump, it just did not report a segfault in dmesg.
Here's the output of `coredumpctl debug`

```shell
root@netdata:/home/max# coredumpctl debug
           PID: 54749 (netdata)
           UID: 998 (netdata)
           GID: 999 (netdata)
        Signal: 6 (ABRT)
     Timestamp: Fri 2022-09-23 07:21:55 UTC (7h ago)
  Command Line: /usr/sbin/netdata -P /run/netdata/netdata.pid -D
    Executable: /usr/sbin/netdata
 Control Group: /system.slice/netdata.service
          Unit: netdata.service
         Slice: system.slice
       Boot ID: 57478a0f0add46fe9da0b80f211a28a5
    Machine ID: 22a3a6ef70f741b0b60db5afec90e682
      Hostname: netdata
       Storage: /var/lib/systemd/coredump/core.netdata.998.57478a0f0add46fe9da0b80f211a28a5.54749.1663917715000000.zst (present)
     Disk Size: 214.9M
       Message: Process 54749 (netdata) of user 998 dumped core.

                Found module linux-vdso.so.1 with build-id: 88536f03982667a41f2be85c4e61e7e8307e2700
                Found module libnss_systemd.so.2 with build-id: a01825ee975c21ee533e6b722d150517aeebb77c
                Found module ld-linux-x86-64.so.2 with build-id: 61ef896a699bb1c2e4e231642b2e1688b2f1a61e
                Found module libc.so.6 with build-id: 69389d485a9793dbe873f0ea2c93e02efaa9aa3d
                Found module libgcc_s.so.1 with build-id: 09c4935b79388431a1248f6a98e00d7dc81b8513
                Found module libm.so.6 with build-id: 27e82301dba6c3f644404d504e1bb1c97894b433
                Found module libstdc++.so.6 with build-id: f57e02bfadacc0c923c82457d5e18e1830b5faea
                Found module libjson-c.so.5 with build-id: cd1ad04f5d85001e965fddf7fe021e16ba7a4ba4
                Found module liblz4.so.1 with build-id: a85971851cd059f1af80d553c8e7170d42ec59a1
                Found module libuv.so.1 with build-id: ff2c8af1d41a623ee738cb5839fb10384ad1c65f
                Found module libuuid.so.1 with build-id: 64c0d0cb22fa2bdeca075a0c0418ba5ff314b220
                Found module libssl.so.3 with build-id: 582183daf1ebb8b1b91e11aec57f615f1cdc4158
                Found module libcrypto.so.3 with build-id: 60d206c3617ae7d35f62345a1431caf7c464033c
                Found module libz.so.1 with build-id: 9b5076d56a29e57e87f8065c7bb26390d118f869
                Found module libelf.so.1 with build-id: 0eaf2d056fb292c3da2d99fa16c13d0ec798f121
                Found module netdata with build-id: 4443371a68cf6909938d5ae56411e9d4656834d8
                Stack trace of thread 63481:
                #0  0x00007f2d8fa07a7c __pthread_kill_implementation (libc.so.6 + 0x96a7c)
                #1  0x00007f2d8f9b3476 __GI_raise (libc.so.6 + 0x42476)
                #2  0x00007f2d8f9997f3 __GI_abort (libc.so.6 + 0x287f3)
                #3  0x00007f2d8f9fa6f6 __libc_message (libc.so.6 + 0x896f6)
                #4  0x00007f2d8fa11d7c malloc_printerr (libc.so.6 + 0xa0d7c)
                #5  0x00007f2d8fa1284c unlink_chunk (libc.so.6 + 0xa184c)
                #6  0x00007f2d8fa13dab _int_free (libc.so.6 + 0xa2dab)
                #7  0x00007f2d8fa1672d __GI___libc_free (libc.so.6 + 0xa572d)
                #8  0x00007f2d8fa059cf start_thread (libc.so.6 + 0x949cf)
                #9  0x00007f2d8fa97a00 __clone3 (libc.so.6 + 0x126a00)

                Stack trace of thread 55067:
                #0  0x00007f2d8fa02197 __futex_abstimed_wait_common64 (libc.so.6 + 0x91197)
                #1  0x00007f2d8fa04ac1 __pthread_cond_wait_common (libc.so.6 + 0x93ac1)
                #2  0x00007f2d8ff1d9ad uv_cond_wait (libuv.so.1 + 0x1e9ad)
                #3  0x00007f2d8ff11086 n/a (libuv.so.1 + 0x12086)
                #4  0x00007f2d8fa05b43 start_thread (libc.so.6 + 0x94b43)
                #5  0x00007f2d8fa97a00 __clone3 (libc.so.6 + 0x126a00)

                Stack trace of thread 54749:
                #0  0x00007f2d8fa56868 __GI___clock_nanosleep (libc.so.6 + 0xe5868)
                #1  0x0000555ffca53e4a sleep_usec (netdata + 0x81e4a)
                #2  0x0000555ffca42591 cancel_main_threads (netdata + 0x70591)
                #3  0x0000555ffca427cc netdata_cleanup_and_exit (netdata + 0x707cc)
                #4  0x0000555ffca44524 signals_handle (netdata + 0x72524)
                #5  0x0000555ffca34f77 main (netdata + 0x62f77)
                #6  0x00007f2d8f99ad90 __libc_start_call_main (libc.so.6 + 0x29d90)
                #7  0x00007f2d8f99ae40 __libc_start_main_impl (libc.so.6 + 0x29e40)
                #8  0x0000555ffca37e55 _start (netdata + 0x65e55)

                Stack trace of thread 54941:
                #0  0x00007f2d8fa02197 __futex_abstimed_wait_common64 (libc.so.6 + 0x91197)
                #1  0x00007f2d8fa076a4 __pthread_clockjoin_ex (libc.so.6 + 0x966a4)
                #2  0x0000555ffca7111b netdata_thread_join (netdata + 0x9f11b)
                #3  0x0000555ffcccd0f0 diskspace_main_cleanup (netdata + 0x2fb0f0)
                #4  0x0000555ffcccd9da diskspace_main (netdata + 0x2fb9da)
                #5  0x0000555ffca70c07 thread_start (netdata + 0x9ec07)
                #6  0x00007f2d8fa05b43 start_thread (libc.so.6 + 0x94b43)
                #7  0x00007f2d8fa97a00 __clone3 (libc.so.6 + 0x126a00)

                Stack trace of thread 55071:
                #0  0x00007f2d8fa02197 __futex_abstimed_wait_common64 (libc.so.6 + 0x91197)
                #1  0x00007f2d8fa04ac1 __pthread_cond_wait_common (libc.so.6 + 0x93ac1)
                #2  0x00007f2d8ff1d9ad uv_cond_wait (libuv.so.1 + 0x1e9ad)
                #3  0x00007f2d8ff11086 n/a (libuv.so.1 + 0x12086)
                #4  0x00007f2d8fa05b43 start_thread (libc.so.6 + 0x94b43)
                #5  0x00007f2d8fa97a00 __clone3 (libc.so.6 + 0x126a00)

                Stack trace of thread 55077:
                #0  0x00007f2d8fa02197 __futex_abstimed_wait_common64 (libc.so.6 + 0x91197)
                #1  0x00007f2d8fa04ac1 __pthread_cond_wait_common (libc.so.6 + 0x93ac1)
                #2  0x00007f2d8ff1d9ad uv_cond_wait (libuv.so.1 + 0x1e9ad)
                #3  0x00007f2d8ff11086 n/a (libuv.so.1 + 0x12086)
                #4  0x00007f2d8fa05b43 start_thread (libc.so.6 + 0x94b43)
                #5  0x00007f2d8fa97a00 __clone3 (libc.so.6 + 0x126a00)

                Stack trace of thread 55074:
                #0  0x00007f2d8fa02197 __futex_abstimed_wait_common64 (libc.so.6 + 0x91197)
                #1  0x00007f2d8fa04ac1 __pthread_cond_wait_common (libc.so.6 + 0x93ac1)
                #2  0x00007f2d8ff1d9ad uv_cond_wait (libuv.so.1 + 0x1e9ad)
                #3  0x00007f2d8ff11086 n/a (libuv.so.1 + 0x12086)
                #4  0x00007f2d8fa05b43 start_thread (libc.so.6 + 0x94b43)
                #5  0x00007f2d8fa97a00 __clone3 (libc.so.6 + 0x126a00)

                Stack trace of thread 55070:
                #0  0x00007f2d8fa02197 __futex_abstimed_wait_common64 (libc.so.6 + 0x91197)
                #1  0x00007f2d8fa04ac1 __pthread_cond_wait_common (libc.so.6 + 0x93ac1)
                #2  0x00007f2d8ff1d9ad uv_cond_wait (libuv.so.1 + 0x1e9ad)
                #3  0x00007f2d8ff11086 n/a (libuv.so.1 + 0x12086)
                #4  0x00007f2d8fa05b43 start_thread (libc.so.6 + 0x94b43)
                #5  0x00007f2d8fa97a00 __clone3 (libc.so.6 + 0x126a00)

                Stack trace of thread 55075:
                #0  0x00007f2d8fa02197 __futex_abstimed_wait_common64 (libc.so.6 + 0x91197)
                #1  0x00007f2d8fa04ac1 __pthread_cond_wait_common (libc.so.6 + 0x93ac1)
                #2  0x00007f2d8ff1d9ad uv_cond_wait (libuv.so.1 + 0x1e9ad)
                #3  0x00007f2d8ff11086 n/a (libuv.so.1 + 0x12086)
                #4  0x00007f2d8fa05b43 start_thread (libc.so.6 + 0x94b43)
                #5  0x00007f2d8fa97a00 __clone3 (libc.so.6 + 0x126a00)

                Stack trace of thread 54938:
                #0  0x00007f2d8fa56868 __GI___clock_nanosleep (libc.so.6 + 0xe5868)
                #1  0x0000555ffca53e4a sleep_usec (netdata + 0x81e4a)
                #2  0x0000555ffcc9787a aclk_main (netdata + 0x2c587a)
                #3  0x0000555ffca70c07 thread_start (netdata + 0x9ec07)
                #4  0x00007f2d8fa05b43 start_thread (libc.so.6 + 0x94b43)
                #5  0x00007f2d8fa97a00 __clone3 (libc.so.6 + 0x126a00)

                Stack trace of thread 55072:
                #0  0x00007f2d8fa02197 __futex_abstimed_wait_common64 (libc.so.6 + 0x91197)
                #1  0x00007f2d8fa04ac1 __pthread_cond_wait_common (libc.so.6 + 0x93ac1)
                #2  0x00007f2d8ff1d9ad uv_cond_wait (libuv.so.1 + 0x1e9ad)
                #3  0x00007f2d8ff11086 n/a (libuv.so.1 + 0x12086)
                #4  0x00007f2d8fa05b43 start_thread (libc.so.6 + 0x94b43)
                #5  0x00007f2d8fa97a00 __clone3 (libc.so.6 + 0x126a00)

                Stack trace of thread 55078:
                #0  0x00007f2d8fa02197 __futex_abstimed_wait_common64 (libc.so.6 + 0x91197)
                #1  0x00007f2d8fa04ac1 __pthread_cond_wait_common (libc.so.6 + 0x93ac1)
                #2  0x00007f2d8ff1d9ad uv_cond_wait (libuv.so.1 + 0x1e9ad)
                #3  0x00007f2d8ff11086 n/a (libuv.so.1 + 0x12086)
                #4  0x00007f2d8fa05b43 start_thread (libc.so.6 + 0x94b43)
                #5  0x00007f2d8fa97a00 __clone3 (libc.so.6 + 0x126a00)

                Stack trace of thread 55080:
                #0  0x00007f2d8fa02197 __futex_abstimed_wait_common64 (libc.so.6 + 0x91197)
                #1  0x00007f2d8fa04ac1 __pthread_cond_wait_common (libc.so.6 + 0x93ac1)
                #2  0x00007f2d8ff1d9ad uv_cond_wait (libuv.so.1 + 0x1e9ad)
                #3  0x00007f2d8ff11086 n/a (libuv.so.1 + 0x12086)
                #4  0x00007f2d8fa05b43 start_thread (libc.so.6 + 0x94b43)
                #5  0x00007f2d8fa97a00 __clone3 (libc.so.6 + 0x126a00)

                Stack trace of thread 55082:
                #0  0x00007f2d8fa02197 __futex_abstimed_wait_common64 (libc.so.6 + 0x91197)
                #1  0x00007f2d8fa04ac1 __pthread_cond_wait_common (libc.so.6 + 0x93ac1)
                #2  0x00007f2d8ff1d9ad uv_cond_wait (libuv.so.1 + 0x1e9ad)
                #3  0x00007f2d8ff11086 n/a (libuv.so.1 + 0x12086)
                #4  0x00007f2d8fa05b43 start_thread (libc.so.6 + 0x94b43)
                #5  0x00007f2d8fa97a00 __clone3 (libc.so.6 + 0x126a00)

                Stack trace of thread 55187:
                #0  0x00007f2d8fa859cc __GI___libc_read (libc.so.6 + 0x1149cc)
                #1  0x00007f2d900bfddd n/a (libcrypto.so.3 + 0xe1ddd)
                #2  0x00007f2d900b02eb n/a (libcrypto.so.3 + 0xd22eb)
                #3  0x00007f2d900b32e5 n/a (libcrypto.so.3 + 0xd52e5)
                #4  0x00007f2d900b3467 BIO_read (libcrypto.so.3 + 0xd5467)
                #5  0x00007f2d8ff8a34c n/a (libssl.so.3 + 0x5034c)
                #6  0x00007f2d8ff8bf19 n/a (libssl.so.3 + 0x51f19)
                #7  0x00007f2d8ff663fc n/a (libssl.so.3 + 0x2c3fc)
                #8  0x00007f2d8ff6ccb7 SSL_read (libssl.so.3 + 0x32cb7)
                #9  0x0000555ffcc870c9 read_stream (netdata + 0x2b50c9)
                #10 0x0000555ffcc875ee streaming_parser (netdata + 0x2b55ee)
                #11 0x0000555ffcc882f8 rrdpush_receive.isra.0 (netdata + 0x2b62f8)
                #12 0x0000555ffcc8887e rrdpush_receiver_thread (netdata + 0x2b687e)
                #13 0x0000555ffca70c07 thread_start (netdata + 0x9ec07)
                #14 0x00007f2d8fa05b43 start_thread (libc.so.6 + 0x94b43)
                #15 0x00007f2d8fa97a00 __clone3 (libc.so.6 + 0x126a00)

                Stack trace of thread 63482:
                #0  0x00007f2d8fa859cc __GI___libc_read (libc.so.6 + 0x1149cc)
                #1  0x00007f2d900bfddd n/a (libcrypto.so.3 + 0xe1ddd)
                #2  0x00007f2d900b02eb n/a (libcrypto.so.3 + 0xd22eb)
                #3  0x00007f2d900b32e5 n/a (libcrypto.so.3 + 0xd52e5)
                #4  0x00007f2d900b3467 BIO_read (libcrypto.so.3 + 0xd5467)
                #5  0x00007f2d8ff8a34c n/a (libssl.so.3 + 0x5034c)
                #6  0x00007f2d8ff8bf19 n/a (libssl.so.3 + 0x51f19)
                #7  0x00007f2d8ff663fc n/a (libssl.so.3 + 0x2c3fc)
                #8  0x00007f2d8ff6ccb7 SSL_read (libssl.so.3 + 0x32cb7)
                #9  0x0000555ffcc870c9 read_stream (netdata + 0x2b50c9)
                #10 0x0000555ffcc875ee streaming_parser (netdata + 0x2b55ee)
                #11 0x0000555ffcc882f8 rrdpush_receive.isra.0 (netdata + 0x2b62f8)
                #12 0x0000555ffcc8887e rrdpush_receiver_thread (netdata + 0x2b687e)
                #13 0x0000555ffca70c07 thread_start (netdata + 0x9ec07)
                #14 0x00007f2d8fa05b43 start_thread (libc.so.6 + 0x94b43)
                #15 0x00007f2d8fa97a00 __clone3 (libc.so.6 + 0x126a00)

                Stack trace of thread 55224:
                #0  0x00007f2d8fa859cc __GI___libc_read (libc.so.6 + 0x1149cc)
                #1  0x00007f2d900bfddd n/a (libcrypto.so.3 + 0xe1ddd)
                #2  0x00007f2d900b02eb n/a (libcrypto.so.3 + 0xd22eb)
                #3  0x00007f2d900b32e5 n/a (libcrypto.so.3 + 0xd52e5)
                #4  0x00007f2d900b3467 BIO_read (libcrypto.so.3 + 0xd5467)
                #5  0x00007f2d8ff8a34c n/a (libssl.so.3 + 0x5034c)
                #6  0x00007f2d8ff8bf19 n/a (libssl.so.3 + 0x51f19)
                #7  0x00007f2d8ff663fc n/a (libssl.so.3 + 0x2c3fc)
                #8  0x00007f2d8ff6ccb7 SSL_read (libssl.so.3 + 0x32cb7)
                #9  0x0000555ffcc870c9 read_stream (netdata + 0x2b50c9)
                #10 0x0000555ffcc875ee streaming_parser (netdata + 0x2b55ee)
                #11 0x0000555ffcc882f8 rrdpush_receive.isra.0 (netdata + 0x2b62f8)
                #12 0x0000555ffcc8887e rrdpush_receiver_thread (netdata + 0x2b687e)
                #13 0x0000555ffca70c07 thread_start (netdata + 0x9ec07)
                #14 0x00007f2d8fa05b43 start_thread (libc.so.6 + 0x94b43)
                #15 0x00007f2d8fa97a00 __clone3 (libc.so.6 + 0x126a00)

                Stack trace of thread 54945:
                #0  0x00007f2d8fa56868 __GI___clock_nanosleep (libc.so.6 + 0xe5868)
                #1  0x0000555ffca53e4a sleep_usec (netdata + 0x81e4a)
                #2  0x0000555ffca54115 heartbeat_next (netdata + 0x82115)
                #3  0x0000555ffcce0d8f netdev_main (netdata + 0x30ed8f)
                #4  0x0000555ffca70c07 thread_start (netdata + 0x9ec07)
                #5  0x00007f2d8fa05b43 start_thread (libc.so.6 + 0x94b43)
                #6  0x00007f2d8fa97a00 __clone3 (libc.so.6 + 0x126a00)

                Stack trace of thread 54751:
                #0  0x00007f2d8fa96fde epoll_wait (libc.so.6 + 0x125fde)
                #1  0x00007f2d8ff244c1 n/a (libuv.so.1 + 0x254c1)
                #2  0x00007f2d8ff0dc88 uv_run (libuv.so.1 + 0xec88)
                #3  0x0000555ffccbcb1e spawn_client (netdata + 0x2eab1e)
                #4  0x00007f2d8fa05b43 start_thread (libc.so.6 + 0x94b43)
                #5  0x00007f2d8fa97a00 __clone3 (libc.so.6 + 0x126a00)

                Stack trace of thread 55069:
                #0  0x00007f2d8fa02197 __futex_abstimed_wait_common64 (libc.so.6 + 0x91197)
                #1  0x00007f2d8fa04ac1 __pthread_cond_wait_common (libc.so.6 + 0x93ac1)
                #2  0x00007f2d8ff1d9ad uv_cond_wait (libuv.so.1 + 0x1e9ad)
                #3  0x00007f2d8ff11086 n/a (libuv.so.1 + 0x12086)
                #4  0x00007f2d8fa05b43 start_thread (libc.so.6 + 0x94b43)
                #5  0x00007f2d8fa97a00 __clone3 (libc.so.6 + 0x126a00)

                Stack trace of thread 55079:
                #0  0x00007f2d8fa02197 __futex_abstimed_wait_common64 (libc.so.6 + 0x91197)
                #1  0x00007f2d8fa04ac1 __pthread_cond_wait_common (libc.so.6 + 0x93ac1)
                #2  0x00007f2d8ff1d9ad uv_cond_wait (libuv.so.1 + 0x1e9ad)
                #3  0x00007f2d8ff11086 n/a (libuv.so.1 + 0x12086)
                #4  0x00007f2d8fa05b43 start_thread (libc.so.6 + 0x94b43)
                #5  0x00007f2d8fa97a00 __clone3 (libc.so.6 + 0x126a00)

                Stack trace of thread 55214:
                #0  0x00007f2d8fa859cc __GI___libc_read (libc.so.6 + 0x1149cc)
                #1  0x00007f2d900bfddd n/a (libcrypto.so.3 + 0xe1ddd)
                #2  0x00007f2d900b02eb n/a (libcrypto.so.3 + 0xd22eb)
                #3  0x00007f2d900b32e5 n/a (libcrypto.so.3 + 0xd52e5)
                #4  0x00007f2d900b3467 BIO_read (libcrypto.so.3 + 0xd5467)
                #5  0x00007f2d8ff8a34c n/a (libssl.so.3 + 0x5034c)
                #6  0x00007f2d8ff8bf19 n/a (libssl.so.3 + 0x51f19)
                #7  0x00007f2d8ff663fc n/a (libssl.so.3 + 0x2c3fc)
                #8  0x00007f2d8ff6ccb7 SSL_read (libssl.so.3 + 0x32cb7)
                #9  0x0000555ffcc870c9 read_stream (netdata + 0x2b50c9)
                #10 0x0000555ffcc875ee streaming_parser (netdata + 0x2b55ee)
                #11 0x0000555ffcc882f8 rrdpush_receive.isra.0 (netdata + 0x2b62f8)
                #12 0x0000555ffcc8887e rrdpush_receiver_thread (netdata + 0x2b687e)
                #13 0x0000555ffca70c07 thread_start (netdata + 0x9ec07)
                #14 0x00007f2d8fa05b43 start_thread (libc.so.6 + 0x94b43)
                #15 0x00007f2d8fa97a00 __clone3 (libc.so.6 + 0x126a00)

                Stack trace of thread 55216:
                #0  0x00007f2d8fa859cc __GI___libc_read (libc.so.6 + 0x1149cc)
                #1  0x00007f2d900bfddd n/a (libcrypto.so.3 + 0xe1ddd)
                #2  0x00007f2d900b02eb n/a (libcrypto.so.3 + 0xd22eb)
                #3  0x00007f2d900b32e5 n/a (libcrypto.so.3 + 0xd52e5)
                #4  0x00007f2d900b3467 BIO_read (libcrypto.so.3 + 0xd5467)
                #5  0x00007f2d8ff8a34c n/a (libssl.so.3 + 0x5034c)
                #6  0x00007f2d8ff8bf19 n/a (libssl.so.3 + 0x51f19)
                #7  0x00007f2d8ff663fc n/a (libssl.so.3 + 0x2c3fc)
                #8  0x00007f2d8ff6ccb7 SSL_read (libssl.so.3 + 0x32cb7)
                #9  0x0000555ffcc870c9 read_stream (netdata + 0x2b50c9)
                #10 0x0000555ffcc875ee streaming_parser (netdata + 0x2b55ee)
                #11 0x0000555ffcc882f8 rrdpush_receive.isra.0 (netdata + 0x2b62f8)
                #12 0x0000555ffcc8887e rrdpush_receiver_thread (netdata + 0x2b687e)
                #13 0x0000555ffca70c07 thread_start (netdata + 0x9ec07)
                #14 0x00007f2d8fa05b43 start_thread (libc.so.6 + 0x94b43)
                #15 0x00007f2d8fa97a00 __clone3 (libc.so.6 + 0x126a00)

                Stack trace of thread 54955:
                #0  0x00007f2d8fa5b559 __waitid (libc.so.6 + 0xea559)
                #1  0x0000555ffcb7e758 pluginsd_worker_thread_cleanup (netdata + 0x1ac758)
                #2  0x0000555ffcb7e4aa pluginsd_worker_thread (netdata + 0x1ac4aa)
                #3  0x0000555ffca70c07 thread_start (netdata + 0x9ec07)
                #4  0x00007f2d8fa05b43 start_thread (libc.so.6 + 0x94b43)
                #5  0x00007f2d8fa97a00 __clone3 (libc.so.6 + 0x126a00)

                Stack trace of thread 54948:
                #0  0x00007f2d8fa56868 __GI___clock_nanosleep (libc.so.6 + 0xe5868)
                #1  0x0000555ffca53e4a sleep_usec (netdata + 0x81e4a)
                #2  0x0000555ffca54115 heartbeat_next (netdata + 0x82115)
                #3  0x0000555ffcccd3bc diskspace_slow_worker (netdata + 0x2fb3bc)
                #4  0x0000555ffca70c07 thread_start (netdata + 0x9ec07)
                #5  0x00007f2d8fa05b43 start_thread (libc.so.6 + 0x94b43)
                #6  0x00007f2d8fa97a00 __clone3 (libc.so.6 + 0x126a00)

                Stack trace of thread 63210:
                #0  0x00007f2d8fa859cc __GI___libc_read (libc.so.6 + 0x1149cc)
                #1  0x00007f2d900bfddd n/a (libcrypto.so.3 + 0xe1ddd)
                #2  0x00007f2d900b02eb n/a (libcrypto.so.3 + 0xd22eb)
                #3  0x00007f2d900b32e5 n/a (libcrypto.so.3 + 0xd52e5)
                #4  0x00007f2d900b3467 BIO_read (libcrypto.so.3 + 0xd5467)
                #5  0x00007f2d8ff8a34c n/a (libssl.so.3 + 0x5034c)
                #6  0x00007f2d8ff8bf19 n/a (libssl.so.3 + 0x51f19)
                #7  0x00007f2d8ff663fc n/a (libssl.so.3 + 0x2c3fc)
                #8  0x00007f2d8ff6ccb7 SSL_read (libssl.so.3 + 0x32cb7)
                #9  0x0000555ffcc870c9 read_stream (netdata + 0x2b50c9)
                #10 0x0000555ffcc875ee streaming_parser (netdata + 0x2b55ee)
                #11 0x0000555ffcc882f8 rrdpush_receive.isra.0 (netdata + 0x2b62f8)
                #12 0x0000555ffcc8887e rrdpush_receiver_thread (netdata + 0x2b687e)
                #13 0x0000555ffca70c07 thread_start (netdata + 0x9ec07)
                #14 0x00007f2d8fa05b43 start_thread (libc.so.6 + 0x94b43)
                #15 0x00007f2d8fa97a00 __clone3 (libc.so.6 + 0x126a00)

                Stack trace of thread 55068:
                #0  0x00007f2d8fa02197 __futex_abstimed_wait_common64 (libc.so.6 + 0x91197)
                #1  0x00007f2d8fa04ac1 __pthread_cond_wait_common (libc.so.6 + 0x93ac1)
                #2  0x00007f2d8ff1d9ad uv_cond_wait (libuv.so.1 + 0x1e9ad)
                #3  0x00007f2d8ff11086 n/a (libuv.so.1 + 0x12086)
                #4  0x00007f2d8fa05b43 start_thread (libc.so.6 + 0x94b43)
                #5  0x00007f2d8fa97a00 __clone3 (libc.so.6 + 0x126a00)

                Stack trace of thread 55204:
                #0  0x00007f2d8fa859cc __GI___libc_read (libc.so.6 + 0x1149cc)
                #1  0x00007f2d900bfddd n/a (libcrypto.so.3 + 0xe1ddd)
                #2  0x00007f2d900b02eb n/a (libcrypto.so.3 + 0xd22eb)
                #3  0x00007f2d900b32e5 n/a (libcrypto.so.3 + 0xd52e5)
                #4  0x00007f2d900b3467 BIO_read (libcrypto.so.3 + 0xd5467)
                #5  0x00007f2d8ff8a34c n/a (libssl.so.3 + 0x5034c)
                #6  0x00007f2d8ff8bf19 n/a (libssl.so.3 + 0x51f19)
                #7  0x00007f2d8ff663fc n/a (libssl.so.3 + 0x2c3fc)
                #8  0x00007f2d8ff6ccb7 SSL_read (libssl.so.3 + 0x32cb7)
                #9  0x0000555ffcc870c9 read_stream (netdata + 0x2b50c9)
                #10 0x0000555ffcc875ee streaming_parser (netdata + 0x2b55ee)
                #11 0x0000555ffcc882f8 rrdpush_receive.isra.0 (netdata + 0x2b62f8)
                #12 0x0000555ffcc8887e rrdpush_receiver_thread (netdata + 0x2b687e)
                #13 0x0000555ffca70c07 thread_start (netdata + 0x9ec07)
                #14 0x00007f2d8fa05b43 start_thread (libc.so.6 + 0x94b43)
                #15 0x00007f2d8fa97a00 __clone3 (libc.so.6 + 0x126a00)

                Stack trace of thread 55073:
                #0  0x00007f2d8fa02197 __futex_abstimed_wait_common64 (libc.so.6 + 0x91197)
                #1  0x00007f2d8fa04ac1 __pthread_cond_wait_common (libc.so.6 + 0x93ac1)
                #2  0x00007f2d8ff1d9ad uv_cond_wait (libuv.so.1 + 0x1e9ad)
                #3  0x00007f2d8ff11086 n/a (libuv.so.1 + 0x12086)
                #4  0x00007f2d8fa05b43 start_thread (libc.so.6 + 0x94b43)
                #5  0x00007f2d8fa97a00 __clone3 (libc.so.6 + 0x126a00)

                Stack trace of thread 55062:
                #0  0x00007f2d8fa859cc __GI___libc_read (libc.so.6 + 0x1149cc)
                #1  0x00007f2d8f9fdcb6 _IO_new_file_underflow (libc.so.6 + 0x8ccb6)
                #2  0x00007f2d8f9fee16 __GI__IO_default_uflow (libc.so.6 + 0x8de16)
                #3  0x00007f2d8f9f149c __GI__IO_getline_info (libc.so.6 + 0x8049c)
                #4  0x00007f2d8f9f0490 _IO_fgets (libc.so.6 + 0x7f490)
                #5  0x0000555ffcc87060 read_stream (netdata + 0x2b5060)
                #6  0x0000555ffcc875ee streaming_parser (netdata + 0x2b55ee)
                #7  0x0000555ffcc882f8 rrdpush_receive.isra.0 (netdata + 0x2b62f8)
                #8  0x0000555ffcc8887e rrdpush_receiver_thread (netdata + 0x2b687e)
                #9  0x0000555ffca70c07 thread_start (netdata + 0x9ec07)
                #10 0x00007f2d8fa05b43 start_thread (libc.so.6 + 0x94b43)
                #11 0x00007f2d8fa97a00 __clone3 (libc.so.6 + 0x126a00)

                Stack trace of thread 55076:
                #0  0x00007f2d8fa02197 __futex_abstimed_wait_common64 (libc.so.6 + 0x91197)
                #1  0x00007f2d8fa04ac1 __pthread_cond_wait_common (libc.so.6 + 0x93ac1)
                #2  0x00007f2d8ff1d9ad uv_cond_wait (libuv.so.1 + 0x1e9ad)
                #3  0x00007f2d8ff11086 n/a (libuv.so.1 + 0x12086)
                #4  0x00007f2d8fa05b43 start_thread (libc.so.6 + 0x94b43)
                #5  0x00007f2d8fa97a00 __clone3 (libc.so.6 + 0x126a00)

                Stack trace of thread 54942:
                #0  0x00007f2d8fa02197 __futex_abstimed_wait_common64 (libc.so.6 + 0x91197)
                #1  0x00007f2d8fa076a4 __pthread_clockjoin_ex (libc.so.6 + 0x966a4)
                #2  0x0000555ffca7111b netdata_thread_join (netdata + 0x9f11b)
                #3  0x0000555ffcccf146 proc_main_cleanup (netdata + 0x2fd146)
                #4  0x0000555ffcccf391 proc_main (netdata + 0x2fd391)
                #5  0x0000555ffca70c07 thread_start (netdata + 0x9ec07)
                #6  0x00007f2d8fa05b43 start_thread (libc.so.6 + 0x94b43)
                #7  0x00007f2d8fa97a00 __clone3 (libc.so.6 + 0x126a00)

                Stack trace of thread 55113:
                #0  0x00007f2d8fa859cc __GI___libc_read (libc.so.6 + 0x1149cc)
                #1  0x00007f2d8f9fdcb6 _IO_new_file_underflow (libc.so.6 + 0x8ccb6)
                #2  0x00007f2d8f9fee16 __GI__IO_default_uflow (libc.so.6 + 0x8de16)
                #3  0x00007f2d8f9f149c __GI__IO_getline_info (libc.so.6 + 0x8049c)
                #4  0x00007f2d8f9f0490 _IO_fgets (libc.so.6 + 0x7f490)
                #5  0x0000555ffcc87060 read_stream (netdata + 0x2b5060)
                #6  0x0000555ffcc875ee streaming_parser (netdata + 0x2b55ee)
                #7  0x0000555ffcc882f8 rrdpush_receive.isra.0 (netdata + 0x2b62f8)
                #8  0x0000555ffcc8887e rrdpush_receiver_thread (netdata + 0x2b687e)
                #9  0x0000555ffca70c07 thread_start (netdata + 0x9ec07)
                #10 0x00007f2d8fa05b43 start_thread (libc.so.6 + 0x94b43)
                #11 0x00007f2d8fa97a00 __clone3 (libc.so.6 + 0x126a00)

                Stack trace of thread 55191:
                #0  0x00007f2d8fa859cc __GI___libc_read (libc.so.6 + 0x1149cc)
                #1  0x00007f2d900bfddd n/a (libcrypto.so.3 + 0xe1ddd)
                #2  0x00007f2d900b02eb n/a (libcrypto.so.3 + 0xd22eb)
                #3  0x00007f2d900b32e5 n/a (libcrypto.so.3 + 0xd52e5)
                #4  0x00007f2d900b3467 BIO_read (libcrypto.so.3 + 0xd5467)
                #5  0x00007f2d8ff8a34c n/a (libssl.so.3 + 0x5034c)
                #6  0x00007f2d8ff8bf19 n/a (libssl.so.3 + 0x51f19)
                #7  0x00007f2d8ff663fc n/a (libssl.so.3 + 0x2c3fc)
                #8  0x00007f2d8ff6ccb7 SSL_read (libssl.so.3 + 0x32cb7)
                #9  0x0000555ffcc870c9 read_stream (netdata + 0x2b50c9)
                #10 0x0000555ffcc875ee streaming_parser (netdata + 0x2b55ee)
                #11 0x0000555ffcc882f8 rrdpush_receive.isra.0 (netdata + 0x2b62f8)
                #12 0x0000555ffcc8887e rrdpush_receiver_thread (netdata + 0x2b687e)
                #13 0x0000555ffca70c07 thread_start (netdata + 0x9ec07)
                #14 0x00007f2d8fa05b43 start_thread (libc.so.6 + 0x94b43)
                #15 0x00007f2d8fa97a00 __clone3 (libc.so.6 + 0x126a00)

                Stack trace of thread 54904:
                #0  0x00007f2d8fa96fde epoll_wait (libc.so.6 + 0x125fde)
                #1  0x00007f2d8ff244c1 n/a (libuv.so.1 + 0x254c1)
                #2  0x00007f2d8ff0dc88 uv_run (libuv.so.1 + 0xec88)
                #3  0x0000555ffcc75e7d rrdeng_worker (netdata + 0x2a3e7d)
                #4  0x00007f2d8fa05b43 start_thread (libc.so.6 + 0x94b43)
                #5  0x00007f2d8fa97a00 __clone3 (libc.so.6 + 0x126a00)

                Stack trace of thread 55203:
                #0  0x00007f2d8fa859cc __GI___libc_read (libc.so.6 + 0x1149cc)
                #1  0x00007f2d900bfddd n/a (libcrypto.so.3 + 0xe1ddd)
                #2  0x00007f2d900b02eb n/a (libcrypto.so.3 + 0xd22eb)
                #3  0x00007f2d900b32e5 n/a (libcrypto.so.3 + 0xd52e5)
                #4  0x00007f2d900b3467 BIO_read (libcrypto.so.3 + 0xd5467)
                #5  0x00007f2d8ff8a34c n/a (libssl.so.3 + 0x5034c)
                #6  0x00007f2d8ff8bf19 n/a (libssl.so.3 + 0x51f19)
                #7  0x00007f2d8ff663fc n/a (libssl.so.3 + 0x2c3fc)
                #8  0x00007f2d8ff6ccb7 SSL_read (libssl.so.3 + 0x32cb7)
                #9  0x0000555ffcc870c9 read_stream (netdata + 0x2b50c9)
                #10 0x0000555ffcc875ee streaming_parser (netdata + 0x2b55ee)
                #11 0x0000555ffcc882f8 rrdpush_receive.isra.0 (netdata + 0x2b62f8)
                #12 0x0000555ffcc8887e rrdpush_receiver_thread (netdata + 0x2b687e)
                #13 0x0000555ffca70c07 thread_start (netdata + 0x9ec07)
                #14 0x00007f2d8fa05b43 start_thread (libc.so.6 + 0x94b43)
                #15 0x00007f2d8fa97a00 __clone3 (libc.so.6 + 0x126a00)

                Stack trace of thread 55081:
                #0  0x00007f2d8fa02197 __futex_abstimed_wait_common64 (libc.so.6 + 0x91197)
                #1  0x00007f2d8fa04ac1 __pthread_cond_wait_common (libc.so.6 + 0x93ac1)
                #2  0x00007f2d8ff1d9ad uv_cond_wait (libuv.so.1 + 0x1e9ad)
                #3  0x00007f2d8ff11086 n/a (libuv.so.1 + 0x12086)
                #4  0x00007f2d8fa05b43 start_thread (libc.so.6 + 0x94b43)
                #5  0x00007f2d8fa97a00 __clone3 (libc.so.6 + 0x126a00)

                Stack trace of thread 55186:
                #0  0x00007f2d8fa859cc __GI___libc_read (libc.so.6 + 0x1149cc)
                #1  0x00007f2d900bfddd n/a (libcrypto.so.3 + 0xe1ddd)
                #2  0x00007f2d900b02eb n/a (libcrypto.so.3 + 0xd22eb)
                #3  0x00007f2d900b32e5 n/a (libcrypto.so.3 + 0xd52e5)
                #4  0x00007f2d900b3467 BIO_read (libcrypto.so.3 + 0xd5467)
                #5  0x00007f2d8ff8a34c n/a (libssl.so.3 + 0x5034c)
                #6  0x00007f2d8ff8bf19 n/a (libssl.so.3 + 0x51f19)
                #7  0x00007f2d8ff663fc n/a (libssl.so.3 + 0x2c3fc)
                #8  0x00007f2d8ff6ccb7 SSL_read (libssl.so.3 + 0x32cb7)
                #9  0x0000555ffcc870c9 read_stream (netdata + 0x2b50c9)
                #10 0x0000555ffcc875ee streaming_parser (netdata + 0x2b55ee)
                #11 0x0000555ffcc882f8 rrdpush_receive.isra.0 (netdata + 0x2b62f8)
                #12 0x0000555ffcc8887e rrdpush_receiver_thread (netdata + 0x2b687e)
                #13 0x0000555ffca70c07 thread_start (netdata + 0x9ec07)
                #14 0x00007f2d8fa05b43 start_thread (libc.so.6 + 0x94b43)
                #15 0x00007f2d8fa97a00 __clone3 (libc.so.6 + 0x126a00)

                Stack trace of thread 55207:
                #0  0x00007f2d8fa022db futex_wait (libc.so.6 + 0x912db)
                #1  0x00007f2d8fa13d43 _int_free (libc.so.6 + 0xa2d43)
                #2  0x00007f2d8fa164d3 __GI___libc_free (libc.so.6 + 0xa54d3)
                #3  0x00007f2d8f9efda7 _IO_deallocate_file (libc.so.6 + 0x7eda7)
                #4  0x0000555ffcc883e7 rrdpush_receive.isra.0 (netdata + 0x2b63e7)
                #5  0x0000555ffcc8887e rrdpush_receiver_thread (netdata + 0x2b687e)
                #6  0x0000555ffca70c07 thread_start (netdata + 0x9ec07)
                #7  0x00007f2d8fa05b43 start_thread (libc.so.6 + 0x94b43)
                #8  0x00007f2d8fa97a00 __clone3 (libc.so.6 + 0x126a00)

GNU gdb (Ubuntu 12.0.90-0ubuntu1) 12.0.90
Copyright (C) 2022 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
Type "show copying" and "show warranty" for details.
This GDB was configured as "x86_64-linux-gnu".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
<https://www.gnu.org/software/gdb/bugs/>.
Find the GDB manual and other documentation resources online at:
    <http://www.gnu.org/software/gdb/documentation/>.

For help, type "help".
Type "apropos word" to search for commands related to "word"...
Reading symbols from /usr/sbin/netdata...
(No debugging symbols found in /usr/sbin/netdata)

warning: Can't open file /usr/sbin/netdata (deleted) during file-backed mapping note processing
[New LWP 63481]
[New LWP 55067]
[New LWP 54749]
[New LWP 54941]
[New LWP 55071]
[New LWP 55077]
[New LWP 55074]
[New LWP 55070]
[New LWP 55075]
[New LWP 54938]
[New LWP 55072]
[New LWP 55078]
[New LWP 55080]
[New LWP 55082]
[New LWP 55187]
[New LWP 63482]
[New LWP 55224]
[New LWP 54945]
[New LWP 54751]
[New LWP 55069]
[New LWP 55079]
[New LWP 55214]
[New LWP 55216]
[New LWP 54955]
[New LWP 54948]
[New LWP 63210]
[New LWP 55068]
[New LWP 55204]
[New LWP 55073]
[New LWP 55062]
[New LWP 55076]
[New LWP 54942]
[New LWP 55113]
[New LWP 55191]
[New LWP 54904]
[New LWP 55203]
[New LWP 55081]
[New LWP 55186]
[New LWP 55207]
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".
Core was generated by `/usr/sbin/netdata -P /run/netdata/netdata.pid -D'.
Program terminated with signal SIGABRT, Aborted.
#0  __pthread_kill_implementation (no_tid=0, signo=6, threadid=139833024869952) at ./nptl/pthread_kill.c:44
44      ./nptl/pthread_kill.c: No such file or directory.
[Current thread is 1 (Thread 0x7f2d69c67640 (LWP 63481))]
(gdb)
```

**Comment by MrZammler** [comment]
Hi @CreeperFace00 

I think there are 2 problems here, one is a crash during shutdown (corrupted double-linked list) but then it should have started correctly (I think the shutdown is part of the update procedure).

The dashboard being inaccessible after the start seems like a locking problem...

**Comment by CreeperFace00** [comment]
I was suspecting that as well, as it only seems to stay up for about a day before I have to restart it again.
I'll pause automatic updates and see if it remains functional.

**Comment by CreeperFace00** [comment]
Yep, disabling the update script appears to have solved that issue. 
Since this looks like a separate problem than the original STREAM_RECEIVER segfault issue, would you like me to submit a new issue to help troubleshoot the update procedure?

**Comment by MrZammler** [comment]
Hi @CreeperFace00. Let's get some more info, and maybe we'll open a new issue.

Can you check please from the system logs, cron or similar, when the updater script was run on that day? (2022-09-23) ?

**Comment by MrZammler** [comment]
I don't think it's the updater's fault (unless it somehow asked netdata to stop when it shouldn't..). From the log file:

1) There is a clean exit at 07:20:02
2) Starts again at 07:20:07 with that days nightly (141), so the updater _must_ have run.
3) Asked to exit (SIGTERM) at 07:21:54 <- who asked it? Maybe the updater?
4) This shutdown caused the crash
5) Starts again with 141 version at 07:22:16

2 -> 3 is not that much time, if you have many children and a big dbengine, it could while cleaning up things weren't yet ready and caused the `corrupted double-linked list`...

**Comment by CreeperFace00** [comment]
I'll check the cron logs. 
This parent node is very large, it's receiving 100,000+ metrics per second from all of the children, though I have not cranked any of the retention settings past their defaults since I haven't gotten it stable yet.
When I started the parent again after my last post 3 days ago, I did have to clear the dbengine as the agent would not start, presumably it had been corrupted.
I grep'd the parent's syslog for netdata entries and uploaded it [https://gist.github.com/CreeperFace00/9740f1cd60b246bfa156590d7acf060f](https://gist.github.com/CreeperFace00/9740f1cd60b246bfa156590d7acf060f)

The corruption could be caused by this notable line here `Sep 27 13:54:16 netdata systemd[1]: netdata.service: State 'stop-sigterm' timed out. Killing.` 
Perhaps it is IO related? I'm moving the parent VM's storage to my ZFS array with an NVMe backed SLOG, then I'll re-enable the update script and see if it happens again.

**Comment by MrZammler** [comment]
The crash happened on a streaming thread (63481) while it was shutting down, but the problem could be somewhere else entirely (corruption) and just manifested there...

I tried to reproduce with some amount of children (30) but with not near as much metrics as your case, but couldn't.

The crash yes, could have corrupted dbengine, really sorry about that... I will continue with more stress tests to see if I can replicate it.

**Comment by CreeperFace00** [comment]
Ok, let me know if there's anything I can do to help out. 
Heck, I could even give you SSH into the parent if you need it to test on. It's a dedicated system for this job, so it's not too big of deal for me.

**Comment by CreeperFace00** [comment]
Hmm, I ran the update script for the first time in a few days and it immediately started segfaulted and became unresponsive.
[syslog dump](https://gist.github.com/CreeperFace00/5064e744ccc465ecb88bcbf8080df7db)
[error.log](https://gist.github.com/CreeperFace00/5c745c6313be02c773c01f9d303cdba3)

**Comment by CreeperFace00** [comment]
Weird. It appears the db corrupted again after the update, it seems to just hang (though it is running?) and doesn't even output anything to the logs until I move the items out of /var/cache/netdata/ and start it again, then resumes working normally.

**Comment by CreeperFace00** [comment]
I think I was able to reproduce the crash and db corruption using `systemctl stop netdata` 
I've done cleared the db folder 3 times so far and stopping the daemon seems to produce the database corruption and agent crashes/hangs/strangeness. The behavior isn't always exactly the same, but clearing the databases and restarting the agent returns it to a stable state until the next restart.

**Comment by MrZammler** [comment]
Hi @CreeperFace00 

Is it easy to test a build of the parent without streaming compression? (Easiest way would be to put `[enable_compression="no"]` in `configure.ac` in line 162). Thanks!

**Comment by CreeperFace00** [comment]
I will try that and let you know.

**Comment by MrZammler** [comment]
Hey @CreeperFace00 !

As discussed, can you please check starting from version >=v1.36.0-202 that the parent behaves ok now?

**Comment by MrZammler** [comment]
Thanks Max!
